kinesis/kinesis.go

```diff
@@ -291,14 +291,13 @@ func (outputPlugin *OutputPlugin) AddRecord(records *[]*kinesis.PutRecordsReques
 	return fluentbit.FLB_OK
 }
 
-// FlushAggregatedRecords must be called after
 // Returns FLB_OK, FLB_RETRY, FLB_ERROR
 func (outputPlugin *OutputPlugin) FlushAggregatedRecords(records *[]*kinesis.PutRecordsRequestEntry) int {
 
 	aggRecord, err := outputPlugin.aggregator.AggregateRecords()
 	if err != nil {
 		logrus.Errorf("[kinesis %d] Failed to aggregate record %v\n", outputPlugin.PluginID, err)
-		return fluentbit.FLB_ERROR
+		return fluentbit.FLB_RETRY
 	}
```

Comment:
Hmm. Not sure about this one... If aggregation fails once, won't it probably fail again the next time?

Reply:
Good point, we could spend more time investigating this, I don't have time right now with higher priority issues. 

If there is any way for it to fail and not fail again then we should always return FLB_RETRY so I think if we don't know then we should return retry.

Reply:
No comment on it in original PR :/ https://github.com/aws/amazon-kinesis-streams-for-fluent-bit/pull/60